### PR TITLE
Vue/future quarters

### PR DESCRIPTION
**Pull request by SarthakSingh31**
Test results are listed on https://jira.cac.washington.edu/browse/MUWM-4822

### CODE REVIEW

File: myuw/static/vue/components/_common/visual_schedule/schedule-tab.vue [hunk 1]
```diff
@@ -180,10 +188,9 @@ export default {
   },
   computed: {
     ...mapState({
-      quarterLastDate: (state) => dayjs(
-          state.termData.lastDay, 'dddd, MMMM D, YYYY',
-      ),
+      quarterLastDate: (state) => dayjs(state.termData.lastDay),
```
Comment (fanglinfang) [inline myuw/static/vue/components/_common/visual_schedule/schedule-tab.vue:191]: How do you feel about adding in props:
 term: {
      type: Object,
      required: true,
    },
and replace state.termData to this.term ?

File: myuw/static/vue/components/_common/visual_schedule/schedule.vue [hunk 2]
```diff
@@ -53,9 +53,14 @@ export default {
     'uw-card': Card,
     'uw-schedule-tab': ScheduleTab,
   },
+  props: {
+    term: {
```
Comment (fanglinfang) [inline myuw/static/vue/components/_common/visual_schedule/schedule.vue:57]: Let's change "term" to "quarter" to match with the quarter attribute in term data.
we want add in ...mapState({
 term: (state) => state.visual_schedule.term,

File: myuw/static/vue/future_quarters.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+vueConf.store.registerModule('textbooks', textbooks);
+vueConf.store.registerModule('visual_schedule', visualSchedule);
+
+vueConf.store.commit('addVarToState', {
+  name: 'futureTerm',
+  value: window.futureTerm,  // "year,quarter,[ab]-term"
+});
+vueConf.store.commit('addVarToState', {
+  name: 'futureTermData',
+  value: window.futureTermData,
```
Comment (fanglinfang) [inline myuw/static/vue/future_quarters.js:26]: Did you want to remove the window.futureTermData and use window.termData on all pages?

File: myuw/templates/future_quarters.html [hunk 4]
```diff
@@ -8,14 +8,6 @@
   <script>
     // TODO: Replace this with a object in the context and a json_script.
```
Comment (fanglinfang) [inline myuw/templates/future_quarters.html:9]: Hi Sarthak, do you still want to replace this with a object in the context?

Comment (SarthakSingh31) [reply myuw/templates/future_quarters.html:9]: not right now, this is very low priority

Comment (fanglinfang) [reply myuw/templates/future_quarters.html:9]: sounds good.

File: myuw/static/vue/future_quarters.js (hunk 3 above)
Comment (SarthakSingh31) [reply myuw/static/vue/future_quarters.js:26]: yeah, removing it. I did a project wide search for it but it didn't turn up

File: myuw/templates/future_quarters.html [hunk 5]
```diff
@@ -1,4 +1,42 @@
 {% extends 'base.html' %}
+
+{% load settings_value %}
+{% load json_script %}
+{% load set_context %}
+
+{% block django_json_data %}
+  <script>
+    // TODO: Replace this with a object in the context and a json_script.
+    window.futureTerm = "{{ future_term }}";
```
Comment (fanglinfang) [inline myuw/templates/future_quarters.html:10]: do we ever use this window.futureTerm?

**Comment by fanglinfang** [comment]
Not sure if the VS change is the cause, the VS card of eight 2013 summer B is crashing.

**Review by fanglinfang — APPROVED** [review]

**Pull request merged**